openedx/core/djangoapps/programs/signals.py

```diff
@@ -140,68 +84,73 @@ class of the object instance that sent this signal
+        course_key(CourseLocator): The course run key of the course run which was updated
+
+    Returns:
+        None
     """
-    # Import here instead of top of file since this module gets imported before the credentials app is loaded, resulting
-    # in a Django deprecation warning.
-    from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
-
-    # Avoid scheduling new tasks if we're not using the Credentials IDA
-    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
-        LOGGER.warning(
-            f"Skipping handling of COURSE_CERT_DATE_CHANGE for course {course_key}. Use of the Credentials service is "
-            "disabled."
-        )
+    if not is_credentials_enabled():
         return
 
     LOGGER.info(f"Handling COURSE_CERT_DATE_CHANGE for course {course_key}")
     # import here, because signal is registered at startup, but items in tasks are not yet loaded
-    from openedx.core.djangoapps.programs.tasks import update_certificate_visible_date_on_course_update
     from openedx.core.djangoapps.programs.tasks import update_certificate_available_date_on_course_update
-    # update the awarded credentials `visible_date` attribute in the Credentials service after a date update
+    from openedx.core.djangoapps.programs.tasks import update_certificate_visible_date_on_course_update
     update_certificate_visible_date_on_course_update.delay(str(course_key))
-    # update the (course) certificate configuration in the Credentials service after a date update
     update_certificate_available_date_on_course_update.delay(str(course_key))
+
+
+@receiver(COURSE_PACING_CHANGED, dispatch_uid="update_credentials_on_pacing_change")
+def handle_course_pacing_change(sender, updated_course_overview, **kwargs):  # pylint: disable=unused-argument
+    """
+    If the pacing of a course run has been updated, we should enqueue the tasks responsible for updating the certificate
+    available date (CAD) stored in the Credentials IDA's internal records. This ensures that we are correctly managing
+    the visibiltiy of certificates on learners' program records.
+
+    Args:
+        updated_course_overview(CourseOverview): The course overview of the course run which was just updated
+
+    Returns:
+        None
+    """
+    if not is_credentials_enabled():
+        return
+
+    course_id = str(updated_course_overview.id)
+    LOGGER.info(f"Handling COURSE_PACING_CHANGED for course {course_id}")
+    # import here, because signal is registered at startup, but items in tasks are not yet loaded
+    from openedx.core.djangoapps.programs.tasks import update_certificate_available_date_on_course_update
+    from openedx.core.djangoapps.programs.tasks import update_certificate_visible_date_on_course_update
+    update_certificate_available_date_on_course_update.delay(course_id)
+    update_certificate_visible_date_on_course_update.delay(course_id)
```

Comment:
### New Behavior

It is imperative for us to queue the tasks responsible for updating the certificate available date on a pacing change too. While the LMS/CMS will handle the updates correctly, and course certificates will be displayed correctly to learners, we are missing a crucial step in queueing the update tasks so that Credentials can keep its internal records up to date.

For example, if a course run was to change from "instructor-paced" with a certificate available date to "self-paced", no other changes are required for the monolith to understand visibility preference with regards to certificate visibility. 

However, that certificate available date will stay in place on Credentials which will obstruct the visibility of the certificates on the program record and cause learner confusion.

Now, a course run that updates pacing _or_ display behavior will trigger the update process.